python/pyarrow/tests/conftest.py

```diff
@@ -311,3 +313,23 @@ def s3_server(s3_connection):
         finally:
             if proc is not None:
                 proc.kill()
+
+
+@pytest.fixture(scope='session')
+def limited_s3_user(request, s3_server):
```

Comment:
Why is this method here instead of `util.py`?

Reply:
Also, why is this a fixture if it's invoked as a regular function?

Reply:
I was trying to pass arguments in the fixture. See this https://stackoverflow.com/a/44701916
The fixture `limited_s3_user` can then be used with both the limited policy and the put only policy.

Reply:
This doesn't answer the question: why do you want it to be a fixture?

Reply:
Sorry, this fixture thing was a little confusing at first. I thought by decorating it as a fixture it should set up the s3 properly with the required permissions. But, that might be the reason of the failing CI. Updated the code where it is no longer a fixture, and moved it to util.py as well. Thanks for the suggestion!